contracts/mint1155_factory.sol

```diff
@@ -0,0 +1,54 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+//import 1155 token contract from Openzeppelin
+
+import "./mint1155.sol";
+import "github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/utils/Counters.sol";
+
+contract PaasPopTickets {
+    using Counters for Counters.Counter;
+    Counters.Counter private _itemIds;
+    address public owner;
+
+    constructor() {}
+
+    event ContractCreated(
+        address newAddress,
+        string metadata,
+        string name,
+        string image,
+        address owner,
+        bool refunded,
+        uint256 maxInfected,
+        uint256 itemID
+    );
+
+    function deployCollection(
+        string memory uri,
+        uint256[] memory ids,
+        uint256[] memory amount,
+        string memory name,
+        string memory image,
+        uint256 maxInfected
+    ) public {
+        _itemIds.increment();
+        uint256 itemId = _itemIds.current();
+
+        // qui chi chiama la funzione (chiunque potenzialmente) diventa owner della factory. Non penso sia corretto proporrei piuttosto che solo l'owner del contratto possa chiamare deployCollection(). A questo punto inizializzerei owner nel costruttore
+        owner = msg.sender;
```

Comment:
Whoever calls this function is the new owner of the marketplace. Is this the desired behaviour? I think there are 2 options:

1) Only the contract owner can call this function
2) the msg.sender becomes the owner of the new NFT collection

In both cases, I would remove this line